packages/plugin-pre/src/Pre.ts

```diff
@@ -0,0 +1,50 @@
+import { JWPlugin, JWPluginConfig } from '../../core/src/JWPlugin';
+import { Loadables } from '../../core/src/JWEditor';
+import { Parser } from '../../plugin-parser/src/Parser';
+import { Renderer } from '../../plugin-renderer/src/Renderer';
+import { PreDomParser } from './PreDomParser';
+import { PreDomRenderer } from './PreDomRenderer';
+import { CommandParams } from '../../core/src/Dispatcher';
+import { VNode, isLeaf } from '../../core/src/VNodes/VNode';
+import { distinct } from '../../utils/src/utils';
+import { PreNode } from './PreNode';
+
+export class Pre<T extends JWPluginConfig = JWPluginConfig> extends JWPlugin<T> {
+    commands = {
+        applyPre: {
+            handler: this.applyPre,
+        },
+    };
+    readonly loadables: Loadables<Parser & Renderer> = {
+        parsers: [PreDomParser],
+        renderers: [PreDomRenderer],
+    };
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Change the formatting of the nodes in given range to Pre.
+     *
+     * @param params
+     */
+    applyPre(params: CommandParams): void {
+        const range = params.context.range;
+        let nodesToConvert: VNode[];
+        if (range.isCollapsed()) {
+            nodesToConvert = [range.start.parent];
+        } else {
+            const selectedLeaves = range.selectedNodes(isLeaf);
+            nodesToConvert = distinct(
+                selectedLeaves.map(leaf => (leaf.atomic ? leaf.parent : leaf)),
+            );
```

Comment:
Nooo, you did that construction again... 😿 Could you please try `targetedNodes` to see if it could match your need ? If it does not then we need to find another way to avoid this leaves + map + parent/leaf + distinct construction.

Reply:
Fail on me for merging `Heading` with this construction then. Would you care to try with `targetedNodes` in `Heading` and update it here ?

Reply:
Separate commit?